# <span style="font-family:Papyrus; font-size:4em; color:Red;">Tartalo</span>

## **Things to Remember:**

1) Read the getting started before reading this write-up.

2) All file paths shown are based on the computer used in this write-up.

3) Use the Resource page/pdf to see a list all websites and programs used in this write-up.

## <font color=red>Tartalo 1</font>

A sample quote for a server order was sent to Amaya around 1:20PM PDT on August 21, 2018. What is the email address of the sender?

### Solution Walk Through:
The SMTP file names are in epoch time. Converting to August 21st close to the time of 1:20 PM results in 1534880000. Use Epoch Converter: https://www.epochconverter.com/timezones?q=1534880000&tz=America%2FLos_Angeles. Due to emails with attachments being larger than emails without attachments, sort by size descending.

In [1]:
Get-ChildItem -Path ..\smtp\alabank -Filter 153488* | Sort-Object Length -Descending 



    Directory: Z:\TF8\Artifacts\smtp\alabank


Mode                LastWriteTime     Length Name                              
----                -------------     ------ ----                              
-----         8/23/2018  10:58 PM      94543 1534885413                        
-----         8/23/2018  10:58 PM      48731 1534887908                        
-----         8/23/2018  10:58 PM      48266 1534886499                        
-----         8/23/2018  10:58 PM       3636 1534885750                        
-----         8/23/2018  10:58 PM       2245 1534884720                        
-----         8/23/2018  10:58 PM       2002 1534888441                        
-----         8/23/2018  10:58 PM       1759 1534883251                        
-----         8/23/2018  10:58 PM       1490 1534886896                        
-----         8/23/2018  10:58 PM       1087 1534880910                        




Read the largest email. This one has an attachment, TCinc_Invoice_20170-4072-00.pdf from the email address trashyourcomputers@tcinc.com

In [2]:
Get-Content ..\smtp\alabank\1534885413 -First 100

Return-Path: <trashyourcomputers@tcinc.com>
Delivered-To: alabank@orko.net
Received: from mail.nimbus.net (unknown [52.95.251.10])
	by mail.orko.net (Postfix) with ESMTPS id 90AA417E82A
	for <alabank@orko.net>; Tue, 21 Aug 2018 14:03:33 -0700 (PDT)
Received: from _ (localhost [127.0.0.1])
	by mail.nimbus.net (Postfix) with ESMTPSA id 8D908181DFD
	for <alabank@orko.net>; Tue, 21 Aug 2018 17:02:08 -0400 (EDT)
MIME-Version: 1.0
Content-Type: multipart/mixed;
 boundary="=_23a9f12f02b21a258733d52cb0faa093"
Date: Tue, 21 Aug 2018 17:02:08 -0400
From: trashyourcomputers@tcinc.com
To: alabank@orko.net
Subject: Re: BUYERS BEWARE!
In-Reply-To: <9745b7e341e093f85322a8e3cd92dd0c@orko.net>
References: <c9de5cb3cc614592a0e2bf4172c05ad5@tcinc.com>
 <bae4fefcb7ace604843f54edd37ef322@orko.net>
 <73b06f11931f0cd03c25732d08b2f76b@tcinc.com>
 <26833fa616675ab8d48fd9c1523457f3@orko.net>
 <c777ffeec1e18ce40fe053a383381883@orko.net>
 <9745b7e341e093f85322a8e3cd92dd0c@orko.net>
Message-ID: <e8ea9d80e057d713d9

#### Answer:  trashyourcomputers@tcinc.com

## <font color=red>Tartalo 2</font>

What PDF editor tool was used to craft this PDF?

### Solution Walk Through:

From the question above, 1534885413 has the attached PDF.

In [3]:
Get-Content ..\smtp\alabank\1534885413 -First 30

Return-Path: <trashyourcomputers@tcinc.com>
Delivered-To: alabank@orko.net
Received: from mail.nimbus.net (unknown [52.95.251.10])
	by mail.orko.net (Postfix) with ESMTPS id 90AA417E82A
	for <alabank@orko.net>; Tue, 21 Aug 2018 14:03:33 -0700 (PDT)
Received: from _ (localhost [127.0.0.1])
	by mail.nimbus.net (Postfix) with ESMTPSA id 8D908181DFD
	for <alabank@orko.net>; Tue, 21 Aug 2018 17:02:08 -0400 (EDT)
MIME-Version: 1.0
Content-Type: multipart/mixed;
 boundary="=_23a9f12f02b21a258733d52cb0faa093"
Date: Tue, 21 Aug 2018 17:02:08 -0400
From: trashyourcomputers@tcinc.com
To: alabank@orko.net
Subject: Re: BUYERS BEWARE!
In-Reply-To: <9745b7e341e093f85322a8e3cd92dd0c@orko.net>
References: <c9de5cb3cc614592a0e2bf4172c05ad5@tcinc.com>
 <bae4fefcb7ace604843f54edd37ef322@orko.net>
 <73b06f11931f0cd03c25732d08b2f76b@tcinc.com>
 <26833fa616675ab8d48fd9c1523457f3@orko.net>
 <c777ffeec1e18ce40fe053a383381883@orko.net>
 <9745b7e341e093f85322a8e3cd92dd0c@orko.net>
Message-ID: <e8ea9d80e057d713d9

---
Carve out the PDF using carve.py. This python script will carve out pieces of an smtp file and dump it in the current directory.

---

In [1]:
python "..\..\carve.py" "..\smtp\alabank\1534885413"

[+] Email part ID 0: None
==> Content Type: multipart/mixed

[+] Email part ID 1: None
==> Content Length in bytes: 2083
==> Content Type: text/plain

[+] Email part ID 2: TCinc_Invoice_20170-4072-00.pdf
==> Content Length in bytes: 67338
==> Content Type: application/pdf



In [2]:
python "..\..\carve.py" "..\smtp\alabank\1534885413" 2

[+] Email part ID 0: None
==> Content Type: multipart/mixed

[+] Email part ID 1: None
==> Content Length in bytes: 2083
==> Content Type: text/plain

[+] Email part ID 2: TCinc_Invoice_20170-4072-00.pdf
==> Content Length in bytes: 67338
==> Content Type: application/pdf

Dumping email part ID 2 with filename TCinc_Invoice_20170-4072-00.pdf...
Successfully dumped file TCinc_Invoice_20170-4072-00.pdf


---
From a google search online, the XML meta tag in PDF files for the editor tool used is <pdf: Producer>. 

---

In [3]:
strings .\TCinc_Invoice_20170-4072-00.pdf | findstr /i Producer

<pdf:Producer>Foxit PhantomPDF Printer Version 9.1.0.0531</pdf:Producer>


Another option is to open the PDF in Adobe Acrobat and check the metadata properties.

#### Answer: Foxit PhantomPDF

## <font color=red>Tartalo 3</font>

What was the URL that one of the PDFs tried to reach out to?

### Solution Walk Through:

In email `1534886896` Amaya emailed Jimmy from Trash Your Computers letting him know that she could not open the pdf that he sent. This was in response to the pdf sent in `1534886499`. The assumption is that maybe she tried logging into the pdf and it reached out to a specific address. There are 3 pcap files from 8/21/2018. Wireshark has a tool called mergecap which can merge pcap files that way they can be viewed together as opposed to individually.

---

In [1]:
mergecap -a -w "08-21-2018.pcap" "..\pcaps\2018-08-21_02-26-57.pcap" "..\pcaps\2018-08-21_12-18-14.pcap" "..\pcaps\2018-08-21_21-25-10.pcap"

---
The merged pcap can then be analyzed to see where the pdf might have tried reaching out to. The below tshark example will display the full uri and time of pcap. 
The filters include:
- IP Source of 192.168.1.11
- Request method of GET
- Frame time >= 8/21/2018 14:20
- Frame time <= 8/21/2018 14:30

The specific frame time was based on the second pdf being sent at 14:21 and Amaya responding at 14:26 in email `1534886896` that she couldn't log in to the pdf. Seems a bit odd that one would 'log in' to a pdf.

---

In [1]:
tshark -r 08-21-2018.pcap -T fields -E separator=' | ' -e http.request.full_uri -e frame.time -Y "ip.src == 192.168.1.11 && http.request.method == GET && frame.time >= \""Aug 21, 2018 14:20:00.000\"" && frame.time <= \""Aug 21, 2018 14:30:00.000\"""

http://12.33.44.77/ Aug 21, 2018 14:20:59.373291000 Pacific Daylight Time
http://acroipm2.adobe.com/18/rdr/ENU/win/nooem/none/consumer/message.zip Aug 21, 2018 14:24:07.048948000 Pacific Daylight Time
http://12.33.55.12/login.jar Aug 21, 2018 14:24:23.974425000 Pacific Daylight Time
http://12.33.55.12/favicon.ico Aug 21, 2018 14:24:24.091582000 Pacific Daylight Time
http://12.33.44.77/ Aug 21, 2018 14:25:59.388505000 Pacific Daylight Time


---
In looking at data above, there is a call to http://12.33.55.12/login.jar on August 21st, 2018 at 14:24 and based on Amaya's email about not being able to log in, this looks to be correct.

---

#### Answer: http://12.33.55.12/login.jar

## <font color=red>Tartalo 4</font>

What is the md5sum of the first PDF sent from Trash Your Computers Inc.?

### Solution Walk Through:
The same file from question 1, has already been carved out of the smtp session.

In [5]:
get-filehash TCinc_Invoice_20170-4072-00.pdf -algorithm md5


Algorithm       Hash                                                           
---------       ----                                                           
MD5             E20FF8395929FD5CF6B8A8417951CC56                               




#### Answer: E20FF8395929FD5CF6B8A8417951CC56

## <font color=red>Tartalo 5</font>

A second email with another PDF was sent soon after the first one. What is the md5sum of the second PDF sent from Trash Your Computers Inc.?

### Solution Walk Through:
Instead of going through the list of big smtp files from the first question to find a PDF, strings on the smtp folder is probably the best bet.

In [10]:
strings -s ..\smtp\alabank | findstr /i "application/pdf"

Z:\TF8\Artifacts\smtp\alabank\1533671668: Content-Type: application/pdf;
Z:\TF8\Artifacts\smtp\alabank\1534885413: Content-Type: application/pdf;
Z:\TF8\Artifacts\smtp\alabank\1534887908: Content-Type: application/pdf;
Z:\TF8\Artifacts\smtp\alabank\1534886499: Content-Type: application/pdf;


---
Given the time of the emails, convert from epoch time, and `1534886499` came after to `1534885413` but before `1534887908`.

---

In [14]:
Get-Content ..\smtp\alabank\1534886499 -First 50

Return-Path: <trashyourcomputers@tcinc.com>
Delivered-To: alabank@orko.net
Received: from mail.nimbus.net (unknown [52.95.251.10])
	by mail.orko.net (Postfix) with ESMTPS id 7CF2717E82A
	for <alabank@orko.net>; Tue, 21 Aug 2018 14:21:39 -0700 (PDT)
Received: from _ (localhost [127.0.0.1])
	by mail.nimbus.net (Postfix) with ESMTPSA id 1FDC5181DF7
	for <alabank@orko.net>; Tue, 21 Aug 2018 17:20:13 -0400 (EDT)
MIME-Version: 1.0
Content-Type: multipart/mixed;
 boundary="=_d39e727b93e8445463c738e317084698"
Date: Tue, 21 Aug 2018 17:20:13 -0400
From: trashyourcomputers@tcinc.com
To: alabank@orko.net
Subject: UPDATE: INVOICE NEW
Message-ID: <daddf622ce31afcb3bdc559e59dc6aec@tcinc.com>
X-Sender: trashyourcomputers@tcinc.com
User-Agent: Roundcube Webmail

--=_d39e727b93e8445463c738e317084698
Content-Transfer-Encoding: 7bit
Content-Type: text/plain; charset=US-ASCII;
 format=flowed

Hi alabank,

Kindly view your new updated invoice. It new better view now. This in 
regards to invoice 1201-19219-

In [15]:
python "..\..\carve.py" "..\smtp\alabank\1534886499"

[+] Email part ID 0: None
==> Content Type: multipart/mixed

[+] Email part ID 1: None
==> Content Length in bytes: 160
==> Content Type: text/plain

[+] Email part ID 2: TCinc_Invoice.pdf
==> Content Length in bytes: 34776
==> Content Type: application/pdf



In [16]:
python "..\..\carve.py" "..\smtp\alabank\1534886499" 2

[+] Email part ID 0: None
==> Content Type: multipart/mixed

[+] Email part ID 1: None
==> Content Length in bytes: 160
==> Content Type: text/plain

[+] Email part ID 2: TCinc_Invoice.pdf
==> Content Length in bytes: 34776
==> Content Type: application/pdf

Dumping email part ID 2 with filename TCinc_Invoice.pdf...
Successfully dumped file TCinc_Invoice.pdf


In [17]:
get-filehash TCinc_Invoice.pdf -algorithm md5


Algorithm       Hash                                                           
---------       ----                                                           
MD5             3955FDD379C2D4612B47E5819BDAFE0B                               




#### Answer: 3955FDD379C2D4612B47E5819BDAFE0B

## <font color=red>Tartalo 6</font>

What is the md5sum of the file downloaded when Amaya clicked on the link in the PDF?

### Solution:
Due to the limitations of tshark, for this question, Wireshark will be used.

In Tartalo 3, the merged pcap file **08-21-2018.pcap** was investigated in tshark. Use this pcap again and filter by IP and the http protocol with the following command: **ip.addr == 12.33.55.12 && http**. From Tartalo 3, the request of interest was http://12.33.55.12/login.jar.

![Tar](pics/Tartalo19.png)

In WireShark, go to follow the http stream of the request of interest.

![tar](pics/Tartalo32.png)

Export the object by clicking on it then clicking `save`:

![tar](pics/Tartalo33.png)

Again, use the Powershell `get-filehash` command on the login.jar file.

In [1]:
get-filehash .\login.jar -algorithm md5


Algorithm       Hash                                                           
---------       ----                                                           
MD5             30FD9A333080A21A46F9E96BC164AE28                               




#### Answer: 30fd9a333080a21a46f9e96bc164ae28

## <font color=red>Tartalo 7</font>

Try to analyze the malware that Amaya downloaded from the PDF. How does the malware encode strings?

### Solution:

Open the **login.jar** file in jd-gui.

![tar](pics/Tartalo35.png)

Looking at the file, notice the **DataExtractor.class**. Examining the class further, the function `public static String decryptString(String input, String key)` is used to decrypt strings and within this function `Base64.getDecoder()` is used. It is assumed that the malware encodes strings in base64.

![tar](pics/Tartalo36.png)

#### Answer: base64

## <font color=red>Tartalo 8</font>

We were able to get a recompiled version of the jar file. What is the key used in the repeating xor?

### Solution:

Download the recompiled jar file given. Open the jar file in jd_gui.

![tar](pics/Tartalo38.png)

Within jd_gui there are several classes in this program. In opening up the classes to examine their purpose, notice that the variable `k` is being referenced several times. In opening the **DataExtractor.class** notice that the variable `k = "cat"` is initalized. This variable is used throughout the DataExtractor.class as well as other classes. Therefore, is is assumed the key is **cat**.

![tar](pics/Tartalo39.png)

#### Answer: cat

## <font color=red>Tartalo 9</font>

What is the name of the unblurred version of the quote that gets opened when the Java code is run? Use the .jar file from Tartalo 8.

### Solution:

Continue in jd_gui with the recompiled jar file.

![tar](pics/Tartalo38.png)

Looking around the program, notice that in **x.class** the program is writing a file to the disk referenced in the string variable **unencrypted** as **Ng8RDQIGGhEABgUrKg8CDAgXBk8EBwc=**. It first unencrypts it using the DataExtracor class and then writes it to disk. 

![tar](pics/Tartalo41.png)

Due to the format of the encrypted string as base64, it can be decoded from base64 then xor'd using the key from **tartalo_8** (CyberChef is used in screenshot below). The decoded string is Unencrypted_Invoice.pdf 

![tar](pics/Tartalo43.png)

#### Answer: Unencrypted_Invoice.pdf

## <font color=red>Tartalo 10</font>

What username is used on the site to host files downloaded by this malware? Use the .jar file from Tartalo 8

### Solution:

Continue in jd_gui with the recompiled jar file.

![tar](pics/Tartalo38.png)

Looking through the encoded there is a string calleds picloc in **dataextractor.class**. 

![tar](pics/Tartalo44.png)

This string leads to a couple different strings but one big encoded string is shown.

![tar](pics/Tartalo45.png)

Use CyberChef and decode with base64 and then xor using the key from **tartalo_8**.

![tar](pics/Tartalo47.png)

#### Answer: johnnywrightfoot12

## <font color=red>Tartalo 11</font>

What is the file extension of the files downloaded by this malware to figure out where to exfiltrate data? Use the .jar file from Tartalo 8

### Solution:

Continue in jd_gui with the recompiled jar file.

![tar](pics/Tartalo38.png)

Look through the encoded strings and you'll come across a string called picloc in **dataextractor.class**. 

![tar](pics/Tartalo44.png)

The second part of the string give a hint because it leads to pictureLocator. After that notice another encoded string **TQsEBgY=**.

![tar](pics/Tartalo48.png)

Go to cyberchef and decode with base64 and then xor using the key from **tartalo_8**. The string is .jpeg.

![tar](pics/Tartalo51.png)

#### Answer: .jpeg

## <font color=red>Tartalo 12</font>

What is the name of the file downloaded by the malware that correctly contains information on where to exfiltrate data?

### Solution:

In Tartalo 6 the pcap **2018-08-21_12-18-14.pcap** was examined. The IP address used by the malware in Tartolo 3 can help refine the results and because the question is asking about a downloaded file it is assumed that it is a GET request. With that knowledge, filter the results with **ip.addr == 12.33.55.12 && http.request.method == GET**.

![tar](pics/Tartalo12-1.png)

Select a packet then export HTTP objects (File>Export Objects>HTTP) and organize by Filename. A file named **0.jpeg** has a hostname of **tartalophotos.com**. Select the **0.jpeg** file and save it.

![tar](pics/Tartalo12-2.PNG)

Find the saved **0.jpeg** file. Right click and select **properties** then **details**. Locate the copyright field and notice that a string of text is placed there.

![tar](pics/Tartalo12-3.png)

This text is most likely base64 encoded because of the **==** at the end. Copy this string and using CyberChef bake it with a base64 recipe. The output doesn't make any sense. 

![tar](pics/Tartalo12-4.png)

The xor key of **cat** in Tartalo 8 needs to be used. In CyberChef bake the string with base64 and xor it with a key of **cat**. The output is then **52.95.251.155/upload_file**. This looks like the address of where to exfiltrate data. It is assumed that **0.jpeg** is the answer. 

![tar](pics/Tartalo12-5.png)


#### Answer: 0.jpeg

## <font color=red>Tartalo 13</font>

What is the field name of either of the two fields which contains the location to exfiltrate data to? Use the .jar file from Tartalo 8.

### Solution:

Since 0.jpeg is the file that contains the information on where to exfiltrate, look at what the jar files are doing after downloading this jpeg. Start by examining the different classes. 

![tar](pics/Tartalo38.png)

Notice that in the **ExifExtractor.class**, it has the function **ipExtract**.

![tar](pics/Tartalo57.png)

Looking at that class, there are two encoded strings. 

![tar](pics/Tartalo58.png)

Once decoded in CyberChef, it is **Co** and **(c)**. This leads to the assumption that the copyright field is used. Looking at the copyright details of **0.jpeg** is a good place to start.

![tar](pics/Tartalo59.png)

Locate the **0.jpeg** file from Tartalo 12 and **right click>Properties>Details**. Notice that the copyright field has the **(c)** copyright symbol that was decoded above. It is assumed that the answer is Copyright.

![tar](pics/Tartalo12-3.png)


#### Answer: Copyright

## <font color=red>Tartalo 14</font>

What is the endpoint of where the malware ex-filtrated data?

### Solution:

In **Tartalo 12**, an encoded string was found, use CyberChef to decode with base64 and then xor with the key of cat. The malware exfiltrated the data to **52.95.251.155/upload_file**.

![tar](pics/Tartalo12-5.png)


#### Answer: 52.95.251.155/upload_file


## <font color=red>Tartalo 15</font>

What is the md5sum of one of the archives that was ex-filtrated?

### Solution:

Knowing that **0.jpeg** came from the pcap file **2018-08-21_12-18-14.pcap**, it is assumed that the malware exfiltrated the data after this time. Open the **2018-08-21_12-18-14.pcap** in WireShark and merge the next three pcaps (encompassing the remainder of 8-21 and all of 8-22) into one pcap file.

![tar](pics/Tartalo15-1.png)

From Tartalo 14 we know the IP address of where the malware uploads the payload. Filter the merged pcap file with **ip.addr == 52.95.251.155 && http.request.method == POST**. 

![Heren](pics/Tartalo63.png)

Select the second packet, **774506**, click the **MIME Multipart Media Encapsulation** drop down, then the **Encapsulated multipart part** drop down, and finally the **media type** drop down. Under **media type** right click and select **Export Packet Bytes...**. 

![Heren](pics/Tartalo64.png)

Run md5sum on the file using Powershell `get-filehash [directory] -algorithm md5` and the answer is **6ab16a8b2fd4c035a4b4f81a8c94253f**.


#### Answer: 6ab16a8b2f4c035a4b4f81a8c94253f